pkg/core/mpt/leaf.go

```diff
@@ -41,13 +40,7 @@ func (n *LeafNode) Bytes() []byte {
 
 // DecodeBinary implements io.Serializable.
 func (n *LeafNode) DecodeBinary(r *io.BinReader) {
-	sz := r.ReadVarUint()
-	if sz > MaxValueLength {
-		r.Err = fmt.Errorf("leaf node value is too big: %d", sz)
-		return
-	}
-	n.value = make([]byte, sz)
-	r.ReadBytes(n.value)
+	n.value = r.ReadVarBytes()
```

Comment:
`ReadVarBytes(MaxValueLength)`?

Reply:
Let's then keep the old code because it returns a more detailed error. I dropped the whole commit.